File: 1. 两数之和/main.cpp
#include <bits/stdc++.h>
using namespace std;

// 哈希表
// 时间复杂度：O(n)
// 空间复杂度：O(n)
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); i++)
        {
            auto iter = m.find(target - nums[i]);
            if (iter != m.end())
                return {iter->second, i};
            else
                m[nums[i]] = i;
        }
        return {};
    }
};

File: 1. 两数之和/main.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(0, len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            else:
                d[target - nums[i]] = i
        return []


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, num in enumerate(nums):
            if target - num in d:
                return [d[target - num], i]
            d[nums[i]] = i
        return []
